bash-preexec.sh

```diff
@@ -332,11 +332,14 @@ __bp_install() {
 
     # Install our hooks in PROMPT_COMMAND to allow our trap to know when we've
     # actually entered something.
-    PROMPT_COMMAND=$'__bp_precmd_invoke_cmd\n'
-    if [[ -n "$existing_prompt_command" ]]; then
-        PROMPT_COMMAND+=${existing_prompt_command}$'\n'
-    fi;
-    PROMPT_COMMAND+='__bp_interactive_mode'
+    PROMPT_COMMAND='__bp_precmd_invoke_cmd'
+    PROMPT_COMMAND+=${existing_prompt_command:+$'\n'$existing_prompt_command}
+    if ((BASH_VERSINFO[0] > 5 || BASH_VERSINFO[0] == 5 && BASH_VERSINFO[1] >= 1)); then
+        PROMPT_COMMAND+=('__bp_interactive_mode')
+    else
+        # shellcheck disable=SC2179 # PROMPT_COMMAND is not an array in bash <= 5.0
+        PROMPT_COMMAND+=$'\n__bp_interactive_mode'
```

Comment:
ShellCheck seems to believe `PROMPT_COMMAND` is an array even in `bash < 5.1` because of the above line `PROMPT_COMMAND+=(...)` for `bash >= 5.1`, so ShellCheck suggests rewriting it to `PROMPT_COMMAND+=($'\n__bp_interactive_mode')`. I disabled this warning SC2179 (and related warnings below, SC2178 and SC2179) because this is a wrong suggestion.